File: parent/pinyougou-web-user/src/main/webapp/js/service/addressService.js
// 定义服务层:
app.service('addressService',function($http){
    // 查询当前用户地址信息
    this.findAddressList=function(){
        return $http.get('../address1/findAddressList.do');
    }
    this.findprovice = function(){
        return $http.get('../address1/findprovice.do');
    }
    this.findcity = function(provinceId){
        return $http.get('../address1/findcity.do?provinceId='+provinceId);
    }
    this.findarea = function(cityId){
        return $http.get('../address1/findarea.do?cityId='+cityId);
    }
    //查询实体
    this.findOne=function(id) {
        return $http.get('../address1/findOne.do?id=' + id);
    }
    // 添加
    this.add=function (entity) {
        return $http.post('../address1/add.do',entity);
    }
    //修改
    this.update=function (entity) {
        return $http.post('../address1/update.do',entity);

    }
    //删除
    this.delete=function (id) {
        return $http.get('../address1/delete.do?id='+id);
    }
    //设为默认
    this.setDefault=function (id) {
        return $http.get('../address1/setDefault.do?id='+id);
    }

});

File: parent/pinyougou-interface/src/main/java/cn/itcast/core/service/AddressService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.address.Address;
import cn.itcast.core.pojo.address.Areas;
import cn.itcast.core.pojo.address.Cities;
import cn.itcast.core.pojo.address.Provinces;

import java.util.List;

public interface AddressService {
    List<Address> findListByLoginUser(String name);

    List<Provinces> findProvice();

    List<Cities> findCity(String provinceId);

    List<Areas> findArea(String cityId);

    Address findOne(Long id);

    void update(Address address);

    void delete(Long id);

    void setDefault(Long id, String name);

    void add(Address address, String name);
}
